File: tests/LoginTest.py
import ParentCase
from ParentCase import on_platforms
from ParentCase import browsers
from ParentCase import ParentCase
import unittest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import TimeoutException
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

@on_platforms(browsers)
class LoginTest(ParentCase):
    def test_login(self):
        driver = self.driver
        # go to the home page
        driver.get("http://127.0.0.1:5981/apps/_design/bell/MyApp/index.html")
        # find the login element
        inputElement = driver.find_element_by_name("login")
        # type in the username
        inputElement.send_keys("admin")
        # find the password element
        inputElement = driver.find_element_by_name("password")
        # type in the password
        inputElement.send_keys("password")
        # submit the form
        inputElement.submit()
        # give it time to get to the next page
        WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "formButton")))
        # print page
        print(driver.current_url)
        # ensure we're logged in
        assert driver.current_url == "http://127.0.0.1:5981/apps/_design/bell/MyApp/index.html#configuration/add"

if __name__ == "__main__":
    unittest.main()

File: tests/login_test.js
/// <reference path="./steps.d.ts" />
Feature('Login');

Scenario('Test for Bad Login', function* (I) {
    //// Bad username
    I.amOnPage('http://127.0.0.1:5981/apps/_design/bell/MyApp/index.html');
    I.waitForVisible('//*[@name="login"]');
    I.fillField('Login', '');
    I.fillField('Password', 'password');
    I.click('Sign In');
    I.wait(1);
    I.seeInPopup("Login or password incorrect.");
    I.acceptPopup();
    // Bad password
    I.fillField('Login', 'admin');
    I.fillField('Password', 'incorrect password');
    I.click('Sign In');
    I.wait(1);
    I.seeInPopup("Login or password incorrect.");
    I.acceptPopup();
    I.wait(5);
    I.login('admin', 'password');
    I.wait(5);

    I.seeInCurrentUrl('#dashboard');
});

Scenario('Test for Successful Login', (I) => {
    I.login('admin', 'password');
    I.seeInCurrentUrl('#dashboard');
});

Scenario('Test for Second Successful Login', (I) => {
    I.login('admin', 'password');
    I.seeInCurrentUrl('#dashboard');
});
